gossip/src/ping_pong.rs

```diff
@@ -217,13 +217,15 @@ impl PingCache {
             None => (false, true),
             Some(t) => {
                 let age = now.saturating_duration_since(*t);
-                // Pop if the pong message has expired.
                 if age > self.ttl {
+                    // The pong message has expired.
                     self.pongs.pop(&node);
+                    (false, true)
+                } else {
+                    // If the pong message is not too recent, generate a new ping
+                    // message to extend remote node verification.
+                    (true, age > self.ttl / 8)
                 }
-                // If the pong message is not too recent, generate a new ping
-                // message to extend remote node verification.
-                (true, age > self.ttl / 8)
```

Comment:
This was intentional.
The idea is that if the pubkey has confirmed it is in control of the IP address, then the next contact should be allowed.

Returning false here can cause excessive traffic and exclude some IP addresses entirely (e.g. low staked nodes), because the first time requires a ping, and by the next time that IP is randomly selected for a request, the previous pong has expired.